### PR TITLE
Refactor most and fewest tackle methods and create two helper methods

### PR DESCRIPTION
**Pull request by GregJMitchell**
This Branch:

- Creates two new helper methods
    - total tackles
    - total_tackles_helper
- Refactors fewest and most tackle methods

### CODE REVIEW

File: lib/season_stats_module.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@ def most_accurate_team(season_id)
       team_hash[team] = (total_goals.to_f / total_shots.to_f)
     end
 
-    accurate_team = team_hash.max_by do |team, shot_percentage|
+    accurate_team = team_hash.max_by do |_team, shot_percentage|
```
Comment (c-bartell) [inline lib/season_stats_module.rb:63]: Is the underscore what rubocop suggests for variables that shadow other variables?

File: lib/season_stats_module.rb [hunk 2]
```diff
@@ -106,45 +106,40 @@ def least_accurate_team(season_id)
...
-          0
-        end
+      tackles_hash[team] = total_tackles_helper(season_id, team)
+    end
+    tackles_hash
+  end
+
+  def total_tackles_helper(season_id, team)
+    total_tackles = game_teams_data_for_season(season_id).sum do |game|
+      if game.team_id == team
```
Comment (c-bartell) [inline lib/season_stats_module.rb:119]: Could we memoize this?

File: lib/season_stats_module.rb [hunk 3]
```diff
@@ -14,33 +14,33 @@ def season_coaches(season_id)
...
     season_coaches(season_id).find_all do |coach|
       total_games = game_teams_data_for_season(season_id).count do |game|
         game.head_coach == coach
       end
       total_wins = game_teams_data_for_season(season_id).count do |game|
-        game.head_coach == coach && game.result == "WIN"
+        game.head_coach == coach && game.result == 'WIN'
       end
-      coaches_hash[coach] = ((total_wins.to_f/total_games.to_f) * 100).round(2)
+      coaches_hash[coach] = ((total_wins.to_f / total_games.to_f) * 100).round(2)
```
Comment (s-steel) [inline lib/season_stats_module.rb:25]: Nice, making it more readable.

**Review by s-steel — APPROVED** [review]
Nice refactoring.  Really cleans it up a lot.

**Review by c-bartell — APPROVED** [review]
Looks good to me.

**Pull request merged**